lychee-lib/src/checker/wikilink/resolver.rs

```diff
@@ -0,0 +1,34 @@
+use crate::{Base, ErrorKind, Uri, checker::wikilink::index::WikilinkIndex};
+use log::trace;
+use std::path::{Path, PathBuf};
+
+#[derive(Clone, Debug)]
+pub(crate) struct WikilinkResolver {
+    checker: WikilinkIndex,
+    fallback_extensions: Vec<String>,
+}
+
+/// Tries to resolve a `WikiLink` by searching for the filename in the `WikilinkIndex`
+/// Returns the path of the found file if found, otherwise an Error
+impl WikilinkResolver {
+    pub(crate) fn new(basedir: Option<Base>, fallback_extensions: Vec<String>) -> Self {
```

Comment:
This also won't be optional anymore, once you refactor the `WikilinkIndex`. E.g.:


```suggestion
    pub(crate) fn new(basedir: Base, fallback_extensions: Vec<String>) -> Self {
```